02_knight_and_princess/dana_stepina/main.cpp

```diff
@@ -0,0 +1,97 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Vertex {
+  const VertexId id = 0;
+  std::vector<int> edges_id;
+};
+struct Edge {
+  const EdgeId id = 0;
+  int vertex_start = 0;
+  int vertex_end = 0;
+};
+
+class Graph {
+ public:
+  void add_edges(std::vector<std::array<int, 2>> graph);
+  void add_vertices();

```

Comment:
Эти функции включают в себя слишком большую ответственность, интерфейс можно упростить, что упростит и логику взаимодействия:
```suggestion
  void add_edge(const VertexId& from_vertex_id, const VertexId& to_vertex_id);
  void add_vertex();
```